File: src/main/java/wanted/market/service/ProductService.java
package wanted.market.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import wanted.market.entity.Product;
import wanted.market.entity.User;
import wanted.market.repository.ProductRepository;
import wanted.market.repository.UserRepository;

import java.util.List;
import java.util.Optional;

@Service
public class ProductService {

    @Autowired
    private ProductRepository productRepository;

    // 상품 목록 조회
    public List<Product> getAllProducts() {
        return productRepository.findAll();
    }

    // 상품 상세조회
    public Optional<Product> getProductById(Long id) {
        return productRepository.findById(id);
    }

    // 상품 등록
    @Transactional
    public void registerProduct(String name, int price) throws Exception {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();

        if (authentication == null || !authentication.isAuthenticated()) {
            throw new Exception("사용자 인증이 필요합니다.");
        }

        WantedUserDetails userDetails = (WantedUserDetails) authentication.getPrincipal();

        User seller = userDetails.getUser();

        Product product = new Product();

        product.setName(name);
        product.setPrice(price);
        product.setSeller(seller);

        productRepository.save(product);
    }
}

File: src/main/kotlin/wanted/market/product/ProductService.kt
package wanted.market.product

import jakarta.transaction.Transactional
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import wanted.market.ErrorCode
import wanted.market.member.MemberService

@Service
@Transactional
class ProductService(@Autowired private val productRepository: ProductRepository,
                     @Autowired private val memberService: MemberService) {

    fun findAllProducts(): List<ProductResponse> {
        return productRepository.findAll()
            .map { product-> ProductResponse.from(product)}

    }

    fun findProduct(id: Long): ProductResponse{
        val product = productRepository.findById(id).orElseThrow{
            throw ProductException(ErrorCode.PRODUCT_NOT_FOUND)
        }
        return ProductResponse.from(product)
    }

    fun saveProduct(saveProductRequest: SaveProductRequest) {
        val member = memberService.findMember(saveProductRequest.memberId)
        val newProduct = Product(saveProductRequest.name, saveProductRequest.price, member)
        productRepository.save(newProduct)
    }
}